spec/system/user_page/staff_info_spec.rb

```diff
@@ -17,4 +17,45 @@
       expect(user_page).to have_warning_messages_path(user)
     end
   end
+
+  context "for flagged posts" do
+    before do
+      ReviewableFlaggedPost
+        .statuses
+        .except(:approved)
+        .keys
+        .map do |status|
+          Fabricate(:reviewable_flagged_post, status: status, target_created_by: user)
+        end
+    end
+
+    context "when there are no approved flagged posts" do
+      it "should not display a flagged-posts staff counter" do
+        expect(user_page.visit(user).staff_info_flagged_posts_counter).to be_nil
```

Comment:
I personally find the this assertion hard to read so perhaps something like the following? 

```suggestion
  user_page.visit(user)

  expect(user_page).to have_no_staff_info_flagged_posts_count
```